core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -21,41 +21,82 @@ import com.unciv.ui.victoryscreen.RankingType
+            for (unique in city.getMatchingUniques(UniqueType.UnhappinessFromPopulationTypePercentageChange))
+                if (city.matchesFilter(unique.params[2]) && unique.params[1] == "Specialists")
+                    yieldStats.happiness -= (unique.params[0].toFloat() / 100f)  // relative val is negative, make positive
+            if (city.civInfo.getHappiness() < 0) yieldStats.happiness *= 2  // double weight for unhappy civilization
+        }
+
+        val surplusFood = city.currentSurplusFood
+        // Apply base weights
+        yieldStats.applyRankingWeights()
+
+        if (surplusFood > 0 && city.avoidGrowth) {
+            yieldStats.food = 0f // don't need more food!
         } else {
-            if (stats.food <= 2 || city.civInfo.getHappiness() > 5) rank += stats.food * 1.2f * foodWeight // food get more value to keep city growing
-            else rank += (2.4f + (stats.food - 2) / 2) * foodWeight // 1.2 point for each food up to 2, from there on half a point
+            if (cityAIFocus != CityFocus.NoFocus && cityAIFocus != CityFocus.FoodFocus && cityAIFocus != CityFocus.ProductionGrowthFocus && cityAIFocus != CityFocus.GoldGrowthFocus) {
+                // Focus on non-food/growth
+                if (surplusFood < 0)
+                    yieldStats.food *= 8 // Starving, need Food, get to 0
+                else
+                    yieldStats.food /= 2
+            } else if (!city.avoidGrowth) {
+                // NoFocus or Food/Growth Focus. Target +2 Food Surplus
+                if (surplusFood < 2)
+                    yieldStats.food *= 8
+                else if (cityAIFocus != CityFocus.FoodFocus)
+                    yieldStats.food /= 2
+                if (city.population.population < 5 && cityAIFocus != CityFocus.FoodFocus)
+                // NoFocus or GoldGrow or ProdGrow, not Avoid Growth, pop < 5. FoodFocus already does this up
+                    yieldStats.food *= 3
+            }
+        }
 
+        if (city.population.population < 5) {
+            // "small city" - we care more about food and less about global problems like gold science and culture 
+            // Food already handled above. Science/Culture have low weights in Stats already
+            yieldStats.gold /= 2 // it's barely worth anything at this point
+        } else {
             if (city.civInfo.gold < 0 && city.civInfo.statsForNextTurn.gold <= 0)
-                rank += stats.gold // we have a global problem
-            else rank += stats.gold / 3 // 3 gold is worse than 2 production
-
-            rank += stats.production
-            rank += stats.science
-            if (city.tiles.size < 12 || city.civInfo.wantsToFocusOn(Victory.Focus.Culture)) {
-                rank += stats.culture
-            } else rank += stats.culture / 2
+                yieldStats.gold *= 2 // We have a global problem
+
+            if (city.tiles.size < 12 || city.civInfo.wantsToFocusOn(Focus.Culture))
```

Comment:
You could look into also taking the `Victory.Focus.Science` and other stats into account

Reply:
Though that might be more something for AI and not for players

Reply:
Done